### PR TITLE
IGNITE-11256 Cluster wide read-only mode.

### PR DESCRIPTION
**Pull request by antonovsergey93**

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/ttl/CacheTtlReadOnlyModeSelfTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.assertCachesReadOnlyMode;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.assertDataStreamerReadOnlyMode;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.cacheConfigurations;
+
+/**
+ *
```
Comment (dspavlov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/ttl/CacheTtlReadOnlyModeSelfTest.java]: Please add Javadoc describing what test does in more details than its name.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/ttl/CacheTtlReadOnlyModeSelfTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ */
+public class CacheTtlReadOnlyModeSelfTest extends GridCommonAbstractTest {
+    /** */
+    private static final int EXPIRATION_TIMEOUT = 10;
```
Comment (dspavlov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/ttl/CacheTtlReadOnlyModeSelfTest.java]: Time unit of timeout is not clear from the field name. Please add Javadoc mentioning time unit.

File: modules/platforms/dotnet/Apache.Ignite.Core.Tests/ApiParity/ClusterParityTest.cs [hunk 3]
```diff
@@ -36,7 +36,8 @@ public class ClusterParityTest
             "isBaselineAutoAdjustEnabled",
             "baselineAutoAdjustEnabled",
             "baselineAutoAdjustTimeout",
-            "baselineAutoAdjustStatus"
+            "baselineAutoAdjustStatus",
+            "readOnly"
```
Comment (dspavlov) [inline modules/platforms/dotnet/Apache.Ignite.Core.Tests/ApiParity/ClusterParityTest.cs]: If readonly is not needed, for now, in .NET, it is probably will be required in the future. Consider creating an issue/discuss this topic at https://lists.apache.org/thread.html/a8a585804578b80545e5d4f862f644857858c68124a1c847b9b2c906@%3Cdev.ignite.apache.org%3E

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeNodeJoinTest.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.assertCachesReadOnlyMode;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.assertDataStreamerReadOnlyMode;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.cacheConfigurations;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.cacheNames;
+
+/**
+ *
```
Comment (dspavlov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeNodeJoinTest.java]: Please add a short description of what this test does.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeNodeJoinTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+ */
+public class ClusterReadOnlyModeNodeJoinTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
```
Comment (dspavlov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeNodeJoinTest.java]: it seems we can make this one-liner

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeNodeJoinTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        grid.cluster().readOnly(true);
+
+        assertCachesReadOnlyMode(true, cacheNames());
+        assertDataStreamerReadOnlyMode(true, cacheNames());
+
+        startGrid(1);
+
+        awaitPartitionMapExchange();
+
+        for(int i=0; i<2; i++)
```
Comment (dspavlov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeNodeJoinTest.java]: Spaces are missing, please reformat code.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeSelfTest.java [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+import org.apache.ignite.transactions.Transaction;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.assertCachesReadOnlyMode;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.assertDataStreamerReadOnlyMode;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.cacheConfigurations;
+import static org.apache.ignite.internal.processors.cache.ClusterReadOnlyModeTestUtils.cacheNames;
+
+/**
+ *
```
Comment (dspavlov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cluster/ClusterReadOnlyModeSelfTest.java]: @antonovsergey93  I appreciate you're using self-descriptive names for tests methods. But I still would like to ask to document suite content in its header. It allows contributors to just read Javadoc header and decide if it is what they were looking for or not.

File: modules/platforms/dotnet/Apache.Ignite.Core.Tests/ApiParity/ClusterParityTest.cs (hunk 3 above)
Comment (antonovsergey93) [reply modules/platforms/dotnet/Apache.Ignite.Core.Tests/ApiParity/ClusterParityTest.cs]: The issue already created https://issues.apache.org/jira/browse/IGNITE-11863

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java [hunk 8]
```diff
@@ -512,10 +534,14 @@ protected void afterStateChangeFinished(IgniteUuid msgId, boolean success) {
     ) {
         DiscoveryDataClusterState state = globalState;
 
-        if (log.isInfoEnabled())
-            U.log(log, "Received " + prettyStr(msg.activate()) + " request with BaselineTopology" +
-                (msg.baselineTopology() == null ? ": null"
-                    : "[id=" + msg.baselineTopology().id() + "]"));
+        if (log.isInfoEnabled()) {
```
Comment (Mmuzaf) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java]: Sergey,

There is no need to add if (log.isInfoEnabled()) the U.log implementation already has it.

Comment (antonovsergey93) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java]: Done!

**Pull request closed without merging**